echopype/convert/set_groups_base.py

```diff
@@ -51,6 +81,8 @@ def _encode_dataarray(da, dtype):
         encoded_data, _, _ = coding.times.encode_cf_datetime(
             da, **read_encoding
         )
+    if da.size == 0:
```

Comment:
@imranmaj This should probably go up top so the computer doesn't spend time trying to `encode_cf_datetime`. Interesting that there's an empty DataArray.

Reply:
Good idea